### PR TITLE
Added FPS, FPM, number of encoded frames, overwrite confirmation and …

### PR DESCRIPTION
**Pull request by kustom666**
…default output to file to encoder_example

Addresses issues #115 and #78 
improvements

### CODE REVIEW

**Comment by nwgat** [comment]
it would be useful to know the total frame count like `Frame 24 of 484`
other than that this looks awesome, keep up the great work
i dont mind it as a patch for the time being either

**Comment by tdaede** [comment]
Computing the total number of frames requires some platform specific code to determine if the input is seekable, so it's a little bit more work, hence why I didn't specify it in the original bug. It'd be great to add in another PR though.

**Comment by kustom666** [comment]
Knowing the total number of frames is possible for almost any container but requires some more work. Tdaede's suggestion of opening another PR for that precise fix seems to be the wisest choice

File: examples/encoder_example.c [hunk 1]
```diff
@@ -635,6 +644,14 @@ int main(int argc, char **argv) {
   }
   /*Assume anything following the options must be a file name.*/
   for (; optind < argc; optind++) id_file(&avin, argv[optind]);
+  if(output_provided == 0){
```
Comment (tdaede) [inline examples/encoder_example.c]: !output_provided

File: examples/encoder_example.c [hunk 2]
```diff
@@ -722,9 +739,11 @@ int main(int argc, char **argv) {
      Main compression loop.*/
   fprintf(stderr, "Compressing...\n");
   video_ready = 0;
+  t0 = clock();
   for (;;) {
     ogg_page video_page;
     double video_time;
+    double video_fps = avin.video_fps_n / avin.video_fps_d;
```
Comment (tdaede) [inline examples/encoder_example.c]: Still think you should actually just count frames.

File: examples/encoder_example.c [hunk 3]
```diff
@@ -750,16 +769,21 @@ int main(int argc, char **argv) {
...
+    t1 = clock();
+    time_spent = (double)(t1 - t0) / CLOCKS_PER_SEC;
     fprintf(stderr,
-     "     %i:%02i:%02i.%02i video: %ikbps          ",
+     "     %i:%02i:%02i.%02i video: %ikbps - Frame %i - %f FPS - %f FPM     ",
      (int)time_base/3600, ((int)time_base/60)%60, (int)time_base % 60,
-     (int)(time_base*100 - (long)time_base*100), video_kbps);
+     (int)(time_base*100 - (long)time_base*100), video_kbps, current_frame_no, 
+     (current_frame_no)/time_spent, 
+     (current_frame_no)/time_spent * 60);
```
Comment (tdaede) [inline examples/encoder_example.c]: Operators with precedence greater than + / - have no spaces. So no spaces around *.

(yes, some of the code around it is old and doesn't follow the style guide)

File: examples/encoder_example.c [hunk 4]
```diff
@@ -750,16 +769,21 @@ int main(int argc, char **argv) {
...
     time_base = video_time;
+    current_frame_no = time_base*video_fps;
     if (interactive) {
       fprintf(stderr, "\r");
     }
     else {
       fprintf(stderr, "\n");
     }
+    t1 = clock();
+    time_spent = (double)(t1 - t0) / CLOCKS_PER_SEC;
```
Comment (tdaede) [inline examples/encoder_example.c]: no spaces around /

File: examples/encoder_example.c [hunk 5]
```diff
@@ -33,6 +33,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.*/
 #include <string.h>
 #include <time.h>
 #include <getopt.h>
+
```
Comment (tdaede) [inline examples/encoder_example.c]: extra space

File: examples/encoder_example.c [hunk 6]
```diff
@@ -750,16 +769,21 @@ int main(int argc, char **argv) {
...
       fprintf(stderr, "\r");
     }
     else {
       fprintf(stderr, "\n");
     }
+    t1 = clock();
+    time_spent = (double)(t1 - t0) / CLOCKS_PER_SEC;
     fprintf(stderr,
-     "     %i:%02i:%02i.%02i video: %ikbps          ",
+     "     %i:%02i:%02i.%02i video: %ikbps - Frame %i - %f FPS - %f FPM     ",
```
Comment (tdaede) [inline examples/encoder_example.c]: Consider limiting the precision of the floats, we don't need 6 decimal points of precision :)

**Comment by tdaede** [comment]
Merged, thanks!

**Pull request closed without merging**